File: iut/TP7_TP8/moyenne2.c
#include <stdio.h>

double moyenne2(double notes[], double coeff[], int nb_notes)
{
    int i;
    double res;
    
    i = 0;
    res = 0;
    while (i != nb_notes)
    {
        res = res + notes[i] * coeff[i];
        i++;
    }
    res = res / nb_notes;
    return res;
}

int main(void)
{
    double notes[] = {15, 18, 12, -3, 20, 0};
    double coeff[] = {1, 2, 3, 4, 1, 6};
    int nb_notes = 6;
    double moyenne;

    moyenne = moyenne2(notes, coeff, nb_notes);
    printf("Moyenne: %f\n", moyenne);
    return 0;
}

File: iut/TP7_TP8/moyenne2.py
def moyenne2(notes, coeff, nb_notes):
    i = 0
    res = 0
    while i != nb_notes:
        res = res + notes[i] * coeff[i]
        i += 1
    res = res / nb_notes
    return res


notes = [15, 18, 12, -3, 20, 0]
coeff = [1, 2, 3, 4, 1, 6]
nb_notes = 6
moyenne = moyenne2(notes, coeff, nb_notes)
print("Moyenne:", moyenne)